src/views/nads/form/utils/utils.ts

```diff
@@ -0,0 +1,116 @@
+import { adjectives, animals, uniqueNamesGenerator } from 'unique-names-generator';
+
+import NetworkAttachmentDefinitionModel from '@kubevirt-ui/kubevirt-api/console/models/NetworkAttachmentDefinitionModel';
+import { k8sCreate } from '@openshift-console/dynamic-plugin-sdk';
+import {
+  IPAMConfig,
+  NetworkAttachmentDefinitionAnnotations,
+  NetworkAttachmentDefinitionConfig,
+  NetworkAttachmentDefinitionKind,
+  RESOURCE_NAME_ANNOTATION,
+} from '@utils/resources/nads/types';
+import { isEmpty } from '@utils/utils';
+
+import { NetworkAttachmentDefinitionFormInput, NetworkTypeKeys } from './types';
+import { networkConsole } from '@utils/utils/utils';
+
+export const generateNADName = (): string => {
+  return `network-${uniqueNamesGenerator({
+    dictionaries: [adjectives, animals],
+    separator: '-',
+  })}`;
+};
```

Comment:
export to a shared util file? probably we will be consumed by others as well - add a prefix as arg